### ISSUE TITLE
[java] Method shouldn't have to be abstract if it is implementing an interface and has a comment inside

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#347]
**Rule Set:** design.html#EmptyMethodInAbstractClassShouldBeAbstract

**Description:** The rule is saying I should make my noop method abstract, but the point is that I don't want inheritors to have to deal with that method.

**Code Sample demonstrating the issue:**

```java
public abstract class Foo implements SomeInterface {
    @Override
    public void onDataChanged() {
        // Noop
    }
}
```

**Running PMD through:** Gradle

**Comment by ryan-gustafson** [comment]
My only concern would be Eclipse tends to create methods like this without batting an eye:

	@Override
	public void add(int index, Object element) {
	    // TODO Auto-generated method stub
	}

Which to me, is not a meaningful comment.

**Comment by SUPERCILEX** [comment]
@ryan-gustafson Well no offense, but you're kinda wrong if you aren't using Intellij. 😄 The simple solution to fix what you said would be to continue complaining about empty methods if the comment inside is a `TODO/FIXME`. Seriously, though the JetBrains toolsuite is the way to go IMO.

**Comment by ryan-gustafson** [comment]
I'd be all over Intellij if it was truly free. I should give it a whirl for PMD development sometime. 😺 

I notice the current 2nd test case would be problematic, as it matches this case.

Some options:
* Switch behavior entirely.
* Provide property to allow comment only properties, default to `false`. Don't bother with tags.
* Provide property to allow comment only bodies, default to `true`, and a tag list property, default to `TODO/FIXME`

**Comment by SUPERCILEX** [comment]
Closing old issues.
